CMSgov/mdct-mcr
# implement long-term descriptive fieldIds (mdct-1858)

## Description

Previously the form field IDs have been tied to their position in the form and section, but this is not ideal because those could change in future iterations of the form. Moving us to long-term unique identifiers related to the data gathered, to help us tie these fields together across variants of the form.

Addresses: https://qmacbis.atlassian.net/browse/MDCT-1858

_Reviewer Note: We currently have the field IDs displaying as error message text to make it easier for product to review. This is also causing automated tests to fail. Please disregard for now and we will change error message behavior back to normal prior to deployment._

### How to test

Very little related to functionality actually changes in this PR. All we need to do is make sure that the report field data being set as part of report creation (in AddEditProgramModal) and report submission (ReviewSubmit) is still being set correctly.
1. Create a program on the dashboard.
2. Confirm that database receives passed field data and Section A "Point of Contact" and "Reporting Period" pages hydrate with the submitted data.
3. Submit program via ReviewSubmit.
4. Confirm that database receives passed field data and Section A "Reporting Period" page hydrates with the submitted data.

### Changed Dependencies

n/a

## Code author checklist

## Reviewer checklist (two different people)
